### PR TITLE
Feature POST & Form-Comment Bootstrap

### PR DESCRIPTION
**Pull request by donpaul999**
- POST tutorial complete
- Frontend update - bootstrap classes for form & comment section
- Add non-empty comments and show them on the main page.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -44,10 +40,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String newMessage = getParameter(request, "text-input", "");
+    if(newMessage != "")
```
Comment (ramaroberto) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:50]: Use !==

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -44,10 +40,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
 
   private String convertToJsonUsingGson(ArrayList<String> messages) {
     Gson gson = new Gson();
     String json = gson.toJson(messages);
     return json;
   }
+
+  private String getParameter(HttpServletRequest request, String name,  String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
```
Comment (ramaroberto) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:63]: Use ===

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -44,10 +40,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+  @Override
```
Comment (ramaroberto) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:47]: Add newline before this line.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -44,10 +40,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (ramaroberto) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:48]: Would you mind adding a comment in a Javadoc style to explain what this function is doing? All public functions should have documentation.

http://www.drjava.org/docs/user/ch10.html

**Review by ramaroberto — CHANGES_REQUESTED** [review]

**Comment by ramaroberto** [comment]
Same as before: Feel free to ignore the comments if you're changing that same piece of code in a following PR; but please ACK them by telling me this.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (donpaul999) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:48]: Solved

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (donpaul999) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:47]: Solved

**Review by ramaroberto — APPROVED** [review]

**Pull request merged**